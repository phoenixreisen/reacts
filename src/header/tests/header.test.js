/**
 * @jest-environment jsdom
 */
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../header.r';
import React from 'react';
import fs from 'fs';

//--- Global Scope Stuff -----

delete window.location;

window.location = {
    pathname: '/',
    protocol: 'http',
    host: 'www.phoenixreisen.com'
};

//--- Tests -----

describe('Header should', () => {

    test('render correctly without params', () => {
        const component = renderer.create(
            <Header />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('show version when given', () => {
        const component = renderer.create(
            <Header version="1.0.0" />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('use given params correctly', () => {
        const component = renderer.create(
            <Header
                url="https://www.test.url.com"
                toggleNav={() => console.log('toggle nav')}
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('have logo & favicon files in folder', () => {
        expect(fs.existsSync(`${__dirname}/../favicon.png`)).toBe(true);
        expect(fs.existsSync(`${__dirname}/../phx.logo.svg`)).toBe(true);
    });
 });