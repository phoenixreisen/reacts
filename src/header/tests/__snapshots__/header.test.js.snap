// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly without params 1`] = `
<article
  className="header"
>
  <a
    href="http//www.phoenixreisen.com"
    target="_self"
    title=""
  >
    <picture
      className="header__logo"
    >
      <img
        src="IMGPATH"
      />
    </picture>
  </a>
</article>
`;

exports[`Header should show version when given 1`] = `
<article
  className="header"
>
  <a
    href="http//www.phoenixreisen.com"
    target="_self"
    title=""
  >
    <picture
      className="header__logo"
    >
      <img
        src="IMGPATH"
      />
    </picture>
  </a>
  <div
    className="header__version"
  >
    1.0.0
  </div>
</article>
`;

exports[`Header should use given params correctly 1`] = `
<article
  className="header"
>
  <a
    href="https://www.test.url.com"
    target="_self"
    title=""
  >
    <picture
      className="header__logo"
    >
      <img
        src="IMGPATH"
      />
    </picture>
  </a>
  <div
    className="header__nav-btn"
  >
    <a
      className="nav-btn noprint"
      onClick={[Function]}
      title="Navigation ein- & ausblenden"
    >
      <i
        className="fas fa-bars"
      />
    </a>
  </div>
</article>
`;
